src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/EnvelopeValidator.java

```diff
@@ -86,21 +88,38 @@ public static void assertEnvelopeContainsOcrDataIfRequired(InputEnvelope envelop
      * @param pdfs     to assert against
      */
     public static void assertEnvelopeHasPdfs(InputEnvelope envelope, List<Pdf> pdfs) {
+        List<String> problems = new ArrayList<>();
+
+        List<String> duplicateFileNames =
```

Comment:
Let's extract this to its separate method maybe

Reply:
done